src/test/java/org/junitpioneer/jupiter/StdIoExtensionTests.java

```diff
@@ -10,84 +10,115 @@
-		void needsAnnotation() {
-			ExecutionResults results = PioneerTestKit
-					.executeTestMethodWithParameterTypes(StdIOExtensionConfigurations.class, "noAnnotation",
-						StdIn.class.getName());
+		@DisplayName("fails if the parameter is StdIn or StdOut but StdIoExtension is not active")
+		void needsExtension() {
+			ExecutionResults results = executeTestMethodWithParameterTypes(
+				NotAnnotatedStdIoExtensionConfiguration.class, "noAnnotation", StdOut.class.getName());
 
 			assertThat(results.firstFailuresThrowable())
 					.isInstanceOf(ParameterResolutionException.class)
-					.hasMessageContainingAll("Can not resolve test method parameter", "Method has to be annotated");
+					.hasMessageStartingWith("No ParameterResolver registered");
 		}
 
 		@Test
-		@DisplayName("resolves parameter for type StdIn and annotation")
-		void goodConfig_stdIn() {
-			ExecutionResults results = PioneerTestKit
-					.executeTestMethodWithParameterTypes(StdIOExtensionConfigurations.class, "resolveStdIn",
-						StdIn.class.getName());
-			assertThat(results.numberOfStartedTests()).isGreaterThan(0);
+		@DisplayName("fails if the parameter is StdIn or StdOut but test method is not annotated with @StdIo")
+		void needsAnnotation() {
+			ExecutionResults results = executeTestMethodWithParameterTypes(StdIoExtensionConfigurations.class,
+				"noAnnotation", StdIn.class.getName());
+
+			assertThat(results.firstFailuresThrowable())
+					// This is because the class is annotated with @ExtendWith
+					.isInstanceOf(ExtensionConfigurationException.class)
+					.hasMessage(
+						format("StdIoExtension is active but no %s annotation was found.", StdIo.class.getName()));
 		}
 
 		@Test
-		@DisplayName("resolves parameter for type StdOut")
-		void goodConfig_stdOut() {
-			ExecutionResults results = PioneerTestKit
-					.executeTestMethodWithParameterTypes(StdIOExtensionConfigurations.class, "resolveStdOut",
-						StdOut.class.getName());
-			assertThat(results.numberOfStartedTests()).isGreaterThan(0);
+		@DisplayName("fails if the method is annotated and has input sources but no StdIn parameter")
+		void needsStdIn() {
+			ExecutionResults results = executeTestMethodWithParameterTypes(StdIoExtensionConfigurations.class,
+				"noStdIn", StdOut.class.getName());
+
+			assertThat(results.firstFailuresThrowable())
+					.isInstanceOf(ExtensionConfigurationException.class)
+					.hasMessage(format(
+						"Method has no %s parameter but input sources were provided in the %s annotation (Did you forget to add a test parameter?).",
+						StdIn.class.getName(), StdIo.class.getName()));
 		}
 
 	}
 
 	@Nested
-	@ExtendWith(StdIOExtension.class)
-	class ExtendWithTests {
+	@DisplayName("with standard configuration ")
+	class StdIoTests {
 
 		@Test
 		@DisplayName("catches the output on the standard out as lines")
+		@StdIo
 		void catchesOut(StdOut out) {
 			app.write();
 
```

Comment:
The method gets no values provided, so the expected result is that `out.capturedLines()` is empty.

Where do the expected lines come from? (I know - from the code itself, but you get what I mean?)

Reply:
You don't provide values to `System.out` - you want to test that your SUT writes to `System.out` what you expect it to.

Reply:
The expected lines come from the `app.write()` method. In it, there are calls to `System.out.println()` - which are redirected to StdOut by the extension.